winhttp Multiple Definition Errors under MinGW

Using a freshly cloned copy of libgit2 and the 0.23.4 tag, I get the following error when I try to compile a static executable using a static build of libgit2 and a static build of wxWidgets 3.0.2:
```
/home/User/Code/boss/lib/libgit2.a(winhttp.c.obj):winhttp.c:(.rdata+0x810): multiple definition of `IID_IInternetSecurityManager'
/usr/x86_64-w64-mingw32/sys-root/mingw/lib/../lib/libuuid.a(lib64_libuuid_a-dxva-uuid.o):(.rdata$IID_IInternetSecurityManager[IID_IInternetSecurityManager]+0x0): first defined here
/home/User/Code/boss/lib/libgit2.a(winhttp.c.obj):winhttp.c:(.rdata+0x820): multiple definition of `CLSID_InternetSecurityManager'
/usr/x86_64-w64-mingw32/sys-root/mingw/lib/../lib/libuuid.a(lib64_libuuid_a-uuid.o):(.rdata$CLSID_InternetSecurityManager[CLSID_InternetSecurityManager]+0x0): first defined here
collect2: error: ld returned 1 exit status
CMakeFiles/boss_gui.dir/build.make:640: recipe for target 'boss_gui.exe' failed
make[2]: *** [boss_gui.exe] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/boss_gui.dir/all' failed
make[1]: *** [CMakeFiles/boss_gui.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```
Changing 
```c++
const CLSID CLSID_InternetSecurityManager = { 0x7B8A2D94, 0x0AC9, 0x11D1,
	{ 0x89, 0x6C, 0x00, 0xC0, 0x4F, 0xB6, 0xBF, 0xC4 } };
```
to
```c++
static const CLSID CLSID_InternetSecurityManager = { 0x7B8A2D94, 0x0AC9, 0x11D1,
	{ 0x89, 0x6C, 0x00, 0xC0, 0x4F, 0xB6, 0xBF, 0xC4 } };
```
brought up this error:
```
/home/User/Code/libgit2/src/transports/winhttp.c:55:20: error: static declaration of 'CLSID_InternetSecurityManager' follows non-static declaration
 static const CLSID CLSID_InternetSecurityManager = { 0x7B8A2D94, 0x0AC9, 0x11D1
                    ^
In file included from /home/User/Code/libgit2/deps/winhttp/urlmon.h:43:0,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/objbase.h:163,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/ole2.h:17,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /usr/x86_64-w64-mingw32/sys-root/mingw/include/winsock2.h:23,
                 from /home/User/Code/libgit2/src/common.h:39,
                 from /home/User/Code/libgit2/src/buffer.h:10,
                 from /home/User/Code/libgit2/src/transports/winhttp.c:12:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/urlmon.h:7348:20: note: previous declaration of 'CLSID_InternetSecurityManager' was here
 EXTERN_C const IID CLSID_InternetSecurityManager;
                    ^
/home/User/Code/libgit2/src/transports/winhttp.c: In function 'certificate_check':
/home/User/Code/libgit2/src/transports/winhttp.c:234:2: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  error = t->owner->certificate_check_cb((git_cert *) &cert, valid, t->connectio
  ^
/home/User/Code/libgit2/src/transports/winhttp.c: In function 'write_chunk':
/home/User/Code/libgit2/src/transports/winhttp.c:493:23: warning: format '%X' expects argument of type 'unsigned int', but argument 3 has type 'size_t {aka long long unsigned int}' [-Wformat=]
  git_buf_printf(&buf, "%X\r\n", len);
                       ^
CMakeFiles/git2.dir/build.make:2968: recipe for target 'CMakeFiles/git2.dir/src/transports/winhttp.c.obj' failed
make[2]: *** [CMakeFiles/git2.dir/src/transports/winhttp.c.obj] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/git2.dir/all' failed
make[1]: *** [CMakeFiles/git2.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

The codebase I'm working on doesn't actually use those symbols and only occurs with wxWidgets in the mix, without the GUI it works fine.

--- a/src/transports/winhttp.c
+++ b/src/transports/winhttp.c
@@ -53,10 +53,15 @@ static const int no_check_cert_flags = SECURITY_FLAG_IGNORE_CERT_CN_INVALID |
 	SECURITY_FLAG_IGNORE_UNKNOWN_CA;
 
 #if defined(__MINGW32__)
-const CLSID CLSID_InternetSecurityManager = { 0x7B8A2D94, 0x0AC9, 0x11D1,
+static const CLSID CLSID_InternetSecurityManager_mingw =
+	{ 0x7B8A2D94, 0x0AC9, 0x11D1,
 	{ 0x89, 0x6C, 0x00, 0xC0, 0x4F, 0xB6, 0xBF, 0xC4 } };
-const IID IID_IInternetSecurityManager = { 0x79EAC9EE, 0xBAF9, 0x11CE,
+static const IID IID_IInternetSecurityManager_mingw =
+	{ 0x79EAC9EE, 0xBAF9, 0x11CE,
 	{ 0x8C, 0x82, 0x00, 0xAA, 0x00, 0x4B, 0xA9, 0x0B } };
+
+# define CLSID_InternetSecurityManager CLSID_InternetSecurityManager_mingw
+# define IID_IInternetSecurityManager IID_IInternetSecurityManager_mingw
 #endif
 
 #define OWNING_SUBTRANSPORT(s) ((winhttp_subtransport *)(s)->parent.subtransport)

